### ISSUE TITLE
Links do not work if app is available only to certain groups

### ISSUE THREAD

**Issue by AntDsl** [SergeyMosin/Appointments#154]
Hello,

I use the application in a school environment where teachers can use the application to create links, but the application is not available to students. To achieve that I restrict the application through the Nextcloud apps page to certain groups.

The problem is that when someone that does not have access to the Appointment app tries to open a public link while logged in to Nextcloud the link does not work and redirects the user to the main page. This is actually the default behavior when a user tries to open a link that belongs to an app that she/he has no access.

If the user is not logged in, or has normal access to the Appointments app, it works correctly. 

Any ideas for a workaround about that? Maybe a built-in per group restriction would help.

For integrating the app into Nextcloud I simply use the Link editor app, and add the public links to make them available where is needed, so users are already logged in by default.

Having the app available to all users is not a big issue, but having a control over that would be nice. I just prefer to have the interface as less cluttered as possible for students, disabling not needed apps.

**Comment by SergeyMosin** [comment]
Yes, there is no default _partial functionality disable_ setting. However, let's say the public booking page would be available as is, but "control/preview" panel would be disabled for "non teachers". The appointment icon would still be shown in the menu for the "non teachers", but when it is clicked it is possible to show a different page instead of the "control/preview" panel.
The question is, what would the NON "control/preview" panel page say, error, warning ?

_Note to self:  Disable the ``StateControler`` for students as well in case there are some Kevin Mitnicks among them._

**Comment by AntDsl** [comment]
First of all, I want to mention that I used the application recently to a great extend and it was a ton of help! It worked without problems, booking more than 200 appointments. 

Since the application does not have a settings page for the administrator, we cant have a custom page or message right? In that case maybe a simple "The application is not available to any of your groups" would suffice.

**Comment by SergeyMosin** [comment]
Available in v1.8.10 via `occ config:app:set` command, example:
```
php occ config:app:set appointments limitToGroups --value '["group1","group2"]'
``` 
Only users in group1 and group2 will have access to the appointments settings 

occ docs: https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/occ_command.html#config-commands-label

![image](https://user-images.githubusercontent.com/10747559/110228069-2d196800-7ecc-11eb-80cc-a68e7f2ea755.png)

**Comment by AntDsl** [comment]
Thank you! this is very helpful, and actually at the right moment, since we have been using the app extensively this month, and I was asked about the particular problem last week.
